src/vm/eventpipe.cpp

```diff
@@ -336,7 +370,15 @@ void EventPipe::Disable()
         s_pBufferManager->WriteAllBuffersToFile(s_pFile, disableTimeStamp);
 
         // Before closing the file, do rundown.
-        s_pConfig->EnableRundown();
+        const unsigned int numRundownProviders = 2;
+        EventPipeProviderConfiguration rundownProviders[] =
+        {
+            { W("Microsoft-Windows-DotNETRuntime"), 0x80020138, static_cast<unsigned int>(EventPipeEventLevel::Verbose) }, // Public provider.
+            { W("Microsoft-Windows-DotNETRuntimeRundown"), 0x80020138, static_cast<unsigned int>(EventPipeEventLevel::Verbose) } // Rundown provider.
```

Comment:
The RuntimeRundown provider is really legacy.   It dates to a time when ETW did not support 'CAPTURE_STATE'.   

I believe the runtime has been plumbed so that you can get all the rundown you need form the normal .NET Runtime provider (by using CAPTURE_STATE), there is an open question whether you should do this automatically on trace end (I vote yes).

In short we should remove the Rundown provider for EventPipe and fix what we need to do to make it work for the the normal .NET provider (which should be something between 0 and minimal).

Reply:
Makes sense to me.  And I agree that the default for capture state should be yes.  I have filed #16030 to track this.

Reply:
Related to this, I am an advocate of making it optional. It would be useful in cases where the user wants smaller tracing files and to be able to shut down EventPipe quickly, such as when they only care about custom events they generate with EventSource.

Reply:
I worried about this scenario as well.  My resolution was that if users stop the SESSION without first disabling the PROVIDER then that PROVIDERS should implicitly do CAPTURE_STATE before ending.  On the other hand if you turn off the provider before stopping the session, then when the session ends the provider is off, and that CAPTURE_STATE should of course do nothing.    This gives the client control over whether rundown happens or not.   Note also that ideally CAPTURE_STATE is per keyword (thus if you don't have Method events on, you don't get Method event CAPTURE_STATE.  I think we do this (but I have not confirmed it).

Reply:
Sure that would work, but I think my suggestion is simpler still.

Reply:
I think that there is a difference between ETW and EventPipe functionality here that may be what is confusing (at least to me).  When you enable EventPipe today (without sessions expose), you can't change anything without disabling EventPipe.  Are you suggesting that when we expose sessions that we should also allow for modification of the configuration while the session is in-use?  If so, your suggestion will work, but I hadn't considered the fact that you can modify an in-use session.

Reply:
Hmm, I see.  Yes ETW allows providers to be turned on and off on the fly.   I must admit that it is a feature that is little used.   

I guess I would defer the issue.  Make it do rundown on session stop by default, and we can add the other things if/when we care.

Reply:
We have the opportunity to make a change here - we're going to have to do more work on sessions, so we can make a call.  By default we have rundown today, so if we do nothing, we will have this same behavior continue.  However, we can choose to go either way, and I can think of scenarios where you might not want CAPTURE_STATE.